### ISSUE TITLE
`MethodError` during type inference on 0.6.0-rc3

### ISSUE THREAD

**Issue by staticfloat** [JuliaLang/julia#22351]
I'm seeing the following error on the official `0.6.0-rc3` linux binaries:

```
WARNING: An error occurred during inference. Type inference is now partially disabled.
Base.MethodError(f=typeof(Core.Inference.convert)(), args=(Base.AssertionError, "invalid age range update"), world=0x0000000000000ac2)
```

Full backtrace available [here](https://gist.github.com/staticfloat/15d3154fe619683859aa662c0640af5b).  I can't seem to reduce the test case much; it's happening during the `include()` of a large corpus of private code, but this bug is triggered only if I `using DocOpt` before loading in the codebase; just running `include()` on the files directly does not trigger the inference bug.  I have the version of Julia and the packages that trigger this bug built into the `staticfloat/julia_type_inference` image.  This seems to be very sensitive to package version and Julia version; if I mount a custom-built version of Julia into the docker container and run it instead of the official `0.6.0-rc3` binary, it first recompiles all the packages, then is able to parse the code without crashing.  If I then (within the same docker container) run the official `0.6.0-rc3` binary, it recompiles the packages again, but then crashes while parsing the code.  This leads me to believe that this combination of package versions _and_ the official 0.6.0-rc3 binary somehow results in this particular error.

I can provide the source code privately to someone if they want to try and debug this.  Pulling the docker image and mounting my code in should be enough to reproduce.

**Comment by JeffBezanson** [comment]
Looks similar to #22125 ?

**Comment by staticfloat** [comment]
Yes, it looks identical to that.  Except that building `0.6.0-rc3` locally on my machine and rebuilding all the relevant packages does not fix the issue, so I wonder if that would fix #22125 as well.

**Comment by marius311** [comment]
Minor note, 0.6RC3 did not in fact fix #22125, so that's consistent with this being the same thing as that.

**Comment by staticfloat** [comment]
I haven't seen this in a really long time, my guess is it's fixed in `v0.6.0`.
